gdcdatamodel/models/qcreport.py

```diff
@@ -0,0 +1,121 @@
+from sqlalchemy import Boolean, Integer, Column, DateTime, String, Text, Enum, ForeignKey, text
+from sqlalchemy.dialects.postgresql import JSONB
+from sqlalchemy.ext.declarative import declarative_base
+from sqlalchemy.orm import relationship, validates
+
+
+Base = declarative_base()
+SEVERITY = Enum("CRITICAL", "WARNING", "PASSED", name="error_severity")
+TEST_RUN_STATUS = Enum("PENDING", "RUNNING", "SUCCESS", "ERROR", "FAILED", name="test_run_status")
+
+
+class TestRun(Base):
+
+    __tablename__ = 'qc_test_runs'
+
+    id = Column(Integer, primary_key=True)
+    project_id = Column(String(64), nullable=False)
+
+    entity_id = Column(String(64), nullable=False)
+    test_type = Column(String(64), nullable=False)
+    is_stale = Column(Boolean(64), nullable=False, default=False)
+
+    # e.g. pending/running/finished
+    status = Column(TEST_RUN_STATUS, default="PENDING", nullable=False)
+
+    test_results = relationship('ValidationResult',
+                                back_populates='test_run',
+                                cascade='all, delete, delete-orphan')
+
+    date_created = Column(
+        DateTime(timezone=True),
+        nullable=False,
+        server_default=text('now()'),
+    )
+
+    last_updated = Column(
+        DateTime(timezone=True),
+        nullable=False,
+        server_default=text('now()'),
+    )
+
+    def __repr__(self):
+        return "<TestRun(id='%d', test_type='%s', status='%s')>" % (
+            self.id, self.test_type, self.status
+        )
+
+    def to_json(self):
+        return {
+            'id': self.id,
+            'project_id': self.project_id,
+            'entity_id': self.entity_id,
+            'test_type': self.test_type,
+            'status': self.status,
+            'date_created': self.date_created.isoformat(),
+            'last_updated': self.last_updated.isoformat()
+        }
+
+
+class ValidationResult(Base):
+
+    __tablename__ = 'qc_validation_results'
+
+    id = Column(Integer, primary_key=True)
+
+    node_id = Column(String(64), nullable=False)
+
+    error_type = Column(String(128), nullable=True, default='')
+
+    node_type = Column(String(128), nullable=False)
```

Comment:
Probably put a comment as to what's expected here, i.e. `Node.label` or `Node._dictionary['Title']` or `Node.name`?

Reply:
I think best is `Node.label`, since we can then directly map to endpoints.